extensions/sql-database-projects/src/models/options/deployOptionsModel.ts

```diff
@@ -7,251 +7,52 @@ import * as mssql from 'mssql';
-			, this.deploymentOptions.commentOutSetVarDeclarations.displayName
-			, this.deploymentOptions.blockWhenDriftDetected.displayName
-			, this.deploymentOptions.blockOnPossibleDataLoss.displayName
-			, this.deploymentOptions.backupDatabaseBeforeChanges.displayName
-			, this.deploymentOptions.allowIncompatiblePlatform.displayName
-			, this.deploymentOptions.allowDropBlockingAssemblies.displayName
-			, this.deploymentOptions.dropConstraintsNotInSource.displayName
-			, this.deploymentOptions.dropDmlTriggersNotInSource.displayName
-			, this.deploymentOptions.dropExtendedPropertiesNotInSource.displayName
-			, this.deploymentOptions.dropIndexesNotInSource.displayName
-			, this.deploymentOptions.ignoreFileAndLogFilePath.displayName
-			, this.deploymentOptions.ignoreExtendedProperties.displayName
-			, this.deploymentOptions.ignoreDmlTriggerState.displayName
-			, this.deploymentOptions.ignoreDmlTriggerOrder.displayName
-			, this.deploymentOptions.ignoreDefaultSchema.displayName
-			, this.deploymentOptions.ignoreDdlTriggerState.displayName
-			, this.deploymentOptions.ignoreDdlTriggerOrder.displayName
-			, this.deploymentOptions.ignoreCryptographicProviderFilePath.displayName
-			, this.deploymentOptions.verifyDeployment.displayName
-			, this.deploymentOptions.ignoreComments.displayName
-			, this.deploymentOptions.ignoreColumnCollation.displayName
-			, this.deploymentOptions.ignoreAuthorizer.displayName
-			, this.deploymentOptions.ignoreAnsiNulls.displayName
-			, this.deploymentOptions.generateSmartDefaults.displayName
-			, this.deploymentOptions.dropStatisticsNotInSource.displayName
-			, this.deploymentOptions.dropRoleMembersNotInSource.displayName
-			, this.deploymentOptions.dropPermissionsNotInSource.displayName
-			, this.deploymentOptions.dropObjectsNotInSource.displayName
-			, this.deploymentOptions.ignoreColumnOrder.displayName
-			, this.deploymentOptions.ignoreTablePartitionOptions.displayName
-			, this.deploymentOptions.doNotEvaluateSqlCmdVariables.displayName
-			, this.deploymentOptions.disableParallelismForEnablingIndexes.displayName
-			, this.deploymentOptions.disableIndexesForDataPhase.displayName
-			, this.deploymentOptions.restoreSequenceCurrentValue.displayName
-			, this.deploymentOptions.rebuildIndexesOfflineForDataPhase.displayName
-			, this.deploymentOptions.preserveIdentityLastValues.displayName
-			, this.deploymentOptions.isAlwaysEncryptedParameterizationEnabled.displayName
-			, this.deploymentOptions.allowExternalLibraryPaths.displayName
-			, this.deploymentOptions.allowExternalLanguagePaths.displayName
-			, this.deploymentOptions.hashObjectNamesInLogs.displayName
-			, this.deploymentOptions.doNotDropWorkloadClassifiers.displayName
-			, this.deploymentOptions.ignoreWorkloadClassifiers.displayName
-			, this.deploymentOptions.ignoreDatabaseWorkloadGroups.displayName
-			, this.deploymentOptions.doNotDropDatabaseWorkloadGroups.displayName].sort();
+		this.deploymentOptions = { ...defaultOptions };
```

Comment:
The values are objects, right? Then this is still going to have the problem where you're creating a new parent object but the properties contained in it are pointing at the same one as the original so any updates you do here will also update the defaultOptions.

If this is intended behavior and you're ok with modifying the original options then I'd suggest just getting rid of the local var completely and doing `constructor(public deploymentOptions: mssql.DeploymentOptions)`

But if you do want to create a new object then you'll need to do a deepClone - see https://github.com/Microsoft/azuredatastudio/blob/2bc6a0cd012de69504061a876070cfa43533fedc/src/vs/base/common/objects.ts for an implementation

Reply:
Actually it doesn't matter if the defaultoptions gets updated, I'm going to remove the local variable and use the defaultOptions variable.